api/client/service/opts.go

```diff
@@ -0,0 +1,372 @@
+package service
+
+import (
+	"fmt"
+	"math/big"
+	"strconv"
+	"time"
+
+	"github.com/docker/docker/opts"
+	runconfigopts "github.com/docker/docker/runconfig/opts"
+	"github.com/docker/docker/volume"
+	"github.com/docker/engine-api/types/swarm"
+	"github.com/docker/go-connections/nat"
+	units "github.com/docker/go-units"
+	"github.com/spf13/cobra"
+)
+
+var (
+	// DefaultScale is the default scale to use for a replicated service
+	DefaultScale uint64 = 1
+)
+
+type int64Value interface {
+	Value() int64
+}
+
+type memBytes int64
+
+func (m *memBytes) String() string {
+	return strconv.FormatInt(m.Value(), 10)
+}
+
+func (m *memBytes) Set(value string) error {
+	val, err := units.RAMInBytes(value)
```

Comment:
In what case would I need to check `err` here? We always return the err, so if there is an error it will be reported. 

This is how all the types in pflag are implemented, ex: https://github.com/spf13/pflag/blob/master/int64.go#L16-L19